src/AVLTree.h

```diff
@@ -0,0 +1,135 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_AVLTREE_H
+#define DATASTRUCTURESWET1_AVLTREE_H
+
+#include "TNode.h"
+
+
+using std::shared_ptr;
+
+template<class Key, class Info>
+class AVLTree {
```

Comment:
is this class not in the same namespace as the rest?
if the problem is the <class>Impl.h file then you can open the same namespace scope there and implement inside it.
i think... 🤔

Reply:
It's a general code file that we might want to use later in other excercises. According to what I heard. I tried to put it in a namespace and it worked, but clion for some reason screwed the formatting whenever I did it. So I decided to not put it in any namespace like we did in the Mtm exercises in UniqureArray.